files/usr/bin/cinnamon-xlet-makepot

```diff
@@ -0,0 +1,283 @@
+    if files_removed is 0:
+        print("Nothing to remove")
+    else:
+        print('removed %i files' % files_removed)
+
+def scan_xlet():
+    parser = argparse.ArgumentParser(description=USAGE_DESCRIPTION, epilog=USAGE_EPILOG, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-j', '--skip-js', action='store_false', dest='js', default=True,
+        help='If this option is not included, javascript files will be scanned for translatable strings.')
+    parser.add_argument('-p', '--skip-python', action='store_false', dest='python', default=True,
+        help='If this option is not included, python files will be scanned for translatable strings.')
+    parser.add_argument('-i', '--install', action='store_true', dest='install', default=False,
+        help=('Compiles and installs any .po files contained in a po folder to the system locale store. '
+              'Use this option to test your translations locally before uploading to Spices. '
+              'It will use the applet, desklet, or extension UUID as the translation domain.'))
+    parser.add_argument('-r', '--remove', action='store_true', dest='remove', default=False,
+        help=('The opposite of install, removes translations from the store. Again, it uses the UUID '
+              'to find the correct files to remove.'))
+    parser.add_argument('-k', '--keyword', dest='keyword', default='_',
+        help='Change the variable name gettext is assigned to in your files. The default is _.')
+    parser.add_argument('-o', '--potfile', dest='out_file', default=None,
+        help=('Use this option to specify the location for the generated .pot file. By default '
+              '<uuid>/po/<uuid>.pot is used. This is where translators are expecting to find the pot '
+              'file, so you should only use this option if you want the pot file elsewhere for your own use.'))
+    parser.add_argument('dir', help='the path to the applet/desklet/extension directory')
+
+    args = parser.parse_args()
+
+    dir = os.path.abspath(args.dir)
+    if not os.path.exists(dir):
+        print('%s does not exist' % dir)
+        quit()
+
+    uuid = os.path.basename(dir)
+
+    if os.path.exists(os.path.join(dir, 'files', uuid)):
+        dir = os.path.join(dir, 'files', uuid)
+
+    if args.install:
+        do_install(uuid, dir)
+        quit()
+
+    if args.remove:
+        do_remove(uuid)
+        quit()
+
+    if args.out_file is not None:
+        pot_path = os.path.abspath(args.out_file)
+    else:
+        pot_path = os.path.join(dir, 'po', uuid + '.pot')
+
+    pwd = os.getcwd()
+    os.chdir(dir)
+    if args.js or args.python:
+        try:
+            output = subprocess.check_output(["xgettext", "--version"])
+        except OSError:
+            print("xgettext not found, you may need to install the gettext package")
+            quit()
+
+    os.makedirs(os.path.basename(pot_path), mode=0o755, exist_ok=True)
```

Comment:
In this line, `dirname` should be used instead of `basename`. Otherwise, a directory with the POT file name is created inside the current working directory.

Reply:
facepalm... I even looked it up to make sure I got the right one.